app/routes/content.js

```diff
@@ -135,23 +135,27 @@ export default Ember.Route.extend(Analytics, ResetScrollMixin, SetupSubmitContro
     },
     actions: {
         error(error, transition) {
-            const slug = transition.params[transition.targetName].preprint_id;
-
-            if (slug.length === 5) {
-                window.location.href = [
-                    window.location.origin,
-                    slug
-                ].join('/');
+            if (error && error.errors && Ember.isArray(error.errors) && error.errors[0].detail === 'The requested node is no longer available.') {
+                this.intermediateTransitionTo('resource-deleted'); // Node containing preprint has been deleted. 410 Gone.
             } else {
-                const path = ['', 'preprints'];
+                const slug = transition.params[transition.targetName].preprint_id;
+
+                if (slug.length === 5) {
```

Comment:
I think so - @binoculars can explain more. I just moved this logic into an else branch, I didn't change anything else.